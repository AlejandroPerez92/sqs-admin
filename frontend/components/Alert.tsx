import { Collapse, IconButton, Alert as MuiAlert } from "@mui/material";
import { Close as CloseIcon } from "@mui/icons-material";
import { AlertProps } from "../types";

const Alert = (props: AlertProps) => {
  return (
    <Collapse in={props.message !== null}>
      <MuiAlert
        severity={props.severity}
        action={
          <IconButton
            aria-label="close"
            color="inherit"
            size="small"
            onClick={() => {
              props.onClose();
            }}
          >
            <CloseIcon fontSize="inherit" />
          </IconButton>
        }
        sx={{ mb: 2 }}
      >
        {props.message}
      </MuiAlert>
    </Collapse>
  );
};

export default Alert;
