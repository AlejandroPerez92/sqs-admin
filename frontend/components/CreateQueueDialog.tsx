import React, { useState } from "react";
import {
  Button,
  Checkbox,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  FormControlLabel,
  FormGroup,
  TextField,
} from "@mui/material";
import { CreateQueueDialogProps, Queue } from "../types";

const CreateQueueDialog = (props: CreateQueueDialogProps) => {
  const [open, setOpen] = useState(false);
  const [queueName, setQueueName] = useState("");
  const [isFifoQueue, enableFifoQueue] = useState(false);

  const handleClickOpen = () => {
    setQueueName("");
    enableFifoQueue(false);
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleChangeQueueName = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    setQueueName(event.target.value);
  };

  const handleChangeFifoSwitch = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    enableFifoQueue(event.target.checked);
  };

  const submitCreateQueue = () => {
    let sanitizedQueueName = queueName;
    if (isFifoQueue && !queueName.endsWith(".fifo")) {
      sanitizedQueueName = `${sanitizedQueueName}.fifo`;
    }
    const newQueue: Queue = {
      QueueName: sanitizedQueueName,
      QueueAttributes: {
        FifoQueue: `${isFifoQueue}`,
      },
    };
    props.onSubmit(newQueue);
    handleClose();
  };

  return (
    <>
      <Button variant="contained" onClick={handleClickOpen}>
        Create Queue
      </Button>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Create new SQS</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please provide a unique queue name.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="Queue-Name"
            type="text"
            fullWidth
            value={queueName}
            onChange={handleChangeQueueName}
            variant="standard"
          />
          <FormGroup>
            <FormControlLabel
              control={
                <Checkbox
                  checked={isFifoQueue}
                  onChange={handleChangeFifoSwitch}
                  inputProps={{ "aria-label": "controlled" }}
                />
              }
              label="FIFO Queue"
            />
          </FormGroup>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose}>Cancel</Button>
          <Button onClick={submitCreateQueue}>Create</Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default CreateQueueDialog;
