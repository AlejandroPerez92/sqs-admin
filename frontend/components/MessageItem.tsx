import React from "react";
import { Card, CardContent, CardHeader } from "@mui/material";
import { SqsMessage } from "../types";
import { JSONTree } from "react-json-tree";

const toLocaleString = (epochTimeStamp: string) => {
  return new Date(parseInt(epochTimeStamp)).toLocaleString();
};

const getJsonOrRawData = (str: string) => {
  try {
    return JSON.parse(str);
  } catch (e) {
    return str;
  }
};

const MessageItem = (props: { data: SqsMessage }) => {
  return (
    <>
      <Card>
        <CardHeader
          title={`MessageId: ${props.data.messageId}`}
          subheader={`Sent on: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.SentTimestamp
          )}, Received at: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.ApproximateFirstReceiveTimestamp
          )}`}
        />
        <CardContent>
          <JSONTree
            data={getJsonOrRawData(props.data.messageBody)}
            keyPath={["message"]}
          />
        </CardContent>
      </Card>
    </>
  );
};

export default MessageItem;
