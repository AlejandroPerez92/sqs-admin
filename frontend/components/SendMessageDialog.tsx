import React, { useState } from "react";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  TextField,
} from "@mui/material";
import { SendMessageDialogProps } from "../types";

const SendMessageDialog = (props: SendMessageDialogProps) => {
  const [open, setOpen] = useState(false);
  const [messageBody, setMessageBody] = useState("");
  const [messageGroupId, setMessageGroupId] = useState("");

  const handleClickOpen = () => {
    setMessageBody("");
    setOpen(true);
    setMessageGroupId("");
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setMessageBody(event.target.value);
  };

  const handleGroupIdChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setMessageGroupId(event.target.value);
  };

  const submitSendRequest = () => {
    props.onSubmit({
      messageBody: messageBody,
      messageGroupId: messageGroupId === "" ? undefined : messageGroupId,
    });
    handleClose();
  };

  return (
    <>
      <Button
        variant="contained"
        disabled={props.disabled}
        onClick={handleClickOpen}
      >
        Send message
      </Button>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Send message</DialogTitle>
        <DialogContent>
          <DialogContentText>Please provide a message body.</DialogContentText>
          <TextField
            multiline
            rows={4}
            autoFocus
            margin="dense"
            id="messageBody"
            label="Message-Body"
            type="text"
            fullWidth
            value={messageBody}
            onChange={handleChange}
            variant="standard"
          />
          <TextField
            margin="dense"
            id="messageGroupId"
            label="Message-Group-Id"
            type="text"
            fullWidth
            required={true}
            value={messageGroupId}
            onChange={handleGroupIdChange}
            style={{
              display: props.queue?.QueueName.endsWith(".fifo")
                ? "flex"
                : "none",
            }}
            variant="standard"
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose}>Cancel</Button>
          <Button onClick={submitSendRequest}>Send</Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default SendMessageDialog;
