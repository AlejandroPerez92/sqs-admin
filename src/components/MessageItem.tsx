import React from "react";
import { Card, CardContent, CardHeader, Typography } from "@mui/material";
import { SqsMessage } from "../types";

const toLocaleString = (epochTimeStamp: string) => {
  return new Date(parseInt(epochTimeStamp)).toLocaleString();
};

const MessageItem = (props: { data: SqsMessage }) => {
  return (
    <>
      <Card>
        <CardHeader
          title={`MessageId: ${props.data.messageId}`}
          subheader={`Sent on: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.SentTimestamp
          )}, Received at: ${toLocaleString(
            // @ts-ignore
            props.data.messageAttributes.ApproximateFirstReceiveTimestamp
          )}`}
        />
        <CardContent>
          <Typography component={"span"} variant="body2">
            {props.data.messageBody}
          </Typography>
        </CardContent>
      </Card>
    </>
  );
};

export default MessageItem;
