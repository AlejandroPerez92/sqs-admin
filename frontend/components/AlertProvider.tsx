import React, { createContext, useContext, useState, ReactNode } from "react";
import Alert from "./Alert";
import { createPortal } from "react-dom";

type Severity = "error" | "warning" | "info" | "success";

interface AlertContextProps {
  showAlert: (message: string, severity: Severity) => void;
}

interface Alert {
  message: string;
  severity: Severity;
}

const AlertContext = createContext<AlertContextProps | undefined>(undefined);

export const AlertProvider: React.FC<{ children: ReactNode }> = ({
                                                                   children
                                                                 }) => {
  const [alerts, setAlerts] = useState<Map<string, Alert>>(new Map());

  const addAlert = (alert: Alert): void => {
    setAlerts(prevAlerts => {
      const newMap = new Map(prevAlerts);
      const key = Math.random().toString();
      newMap.set(key, alert);
      setTimeout(() => deleteAlert(key.toString()), 3000);

      return newMap;
    });
  };

  const deleteAlert = (key: string): void => {
    setAlerts(prevAlerts => {
      const newMap = new Map(prevAlerts);
      newMap.delete(key);
      return newMap;
    });
  };

  const showAlert = (message: string, severity: Severity) => {
    addAlert({ message, severity });
  };

  return (
    <AlertContext.Provider value={{ showAlert }}>
      {children}
      {createPortal(
        <div
          style={{
            position: "fixed",
            top: 20,
            left: "50%",
            transform: "translateX(-50%)",
            zIndex: 9999,
            width: "90%",
            maxWidth: 500,
            pointerEvents: "none",
          }}
        >
          {Array.from(alerts.entries()).map(([key, alert]) => (
          <Alert
            message={alert.message}
            severity={alert.severity}
            onClose={() => deleteAlert(key)}
            key={key}
          />
          ))}
        </div>,
        document.body
      )}
    </AlertContext.Provider>
  );
};

export const useAlert = (): AlertContextProps => {
  const context = useContext(AlertContext);
  if (!context) {
    throw new Error("useAlert must be used inside AlterProvider wrapper");
  }
  return context;
};
