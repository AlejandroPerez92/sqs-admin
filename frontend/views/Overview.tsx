import React, { useState, useEffect } from "react";
import TabPanel from "../components/TabPanel";
import {
  Alert as MuiAlert,
  AlertTitle,
  Box,
  Button,
  Container,
  Grid,
  Paper,
  Tab,
  Tabs,
  Typography,
} from "@mui/material";
import { Queue, SqsMessage } from "../types";
import CreateQueueDialog from "../components/CreateQueueDialog";
import Alert from "../components/Alert";
import useInterval from "../hooks/useInterval";
import SendMessageDialog from "../components/SendMessageDialog";
import { callApi } from "../api/Http";
import MessageItem from "../components/MessageItem";

const a11yProps = (id: string, index: number) => {
  return {
    key: `queue-${id}-${index}`,
    "aria-controls": `queue-${id}-${index}`,
  };
};

const Overview = () => {
  const [tabIndex, setTabIndex] = useState(0);
  const [queues, setQueues] = useState([] as Queue[]);
  const [messages, setMessages] = useState([] as SqsMessage[]);
  const [reload, triggerReload] = useState(true);
  const [error, setError] = useState("");
  const [disabledStatus, setDisabledStatus] = useState(true);

  useInterval(async () => {
    await receiveMessageFromCurrentQueue();
  }, 3000);

  useEffect(() => {
    receiveMessageFromCurrentQueue();
    // eslint-disable-next-line
  }, [queues, tabIndex]);

  useEffect(() => {
    callApi({
      method: "GET",
      onSuccess: (data: Queue[]) => {
        setQueues(data);
        if (data.length > 0) {
          setTabIndex(data.length - 1);
          setDisabledStatus(false);
        } else {
          setTabIndex(0);
          setDisabledStatus(true);
        }
      },
      onError: setError,
    });
  }, [reload]);

  const handleChange = (event: React.SyntheticEvent, newTabIndex: number) => {
    setTabIndex(newTabIndex);
  };

  const receiveMessageFromCurrentQueue = async () => {
    let queueUrl = queues[tabIndex]?.QueueUrl || null;
    if (queueUrl != null) {
      await callApi({
        method: "POST",
        action: "GetMessages",
        queue: queues[tabIndex],
        onSuccess: setMessages,
        onError: setError,
      });
    }
  };

  const createNewQueue = async (queue: Queue) => {
    await callApi({
      method: "POST",
      action: "CreateQueue",
      queue: queue,
      onSuccess: () => {
        setTimeout(() => {
          triggerReload(!reload);
        }, 1000);
      },
      onError: setError,
    });
  };

  const purgeCurrentQueue = async () => {
    await callApi({
      method: "POST",
      action: "PurgeQueue",
      queue: queues[tabIndex],
      onSuccess: () => {
        setMessages([]);
      },
      onError: setError,
    });
  };

  const deleteCurrentQueue = async () => {
    await callApi({
      method: "POST",
      action: "DeleteQueue",
      queue: queues[tabIndex],
      onSuccess: () => {
        setMessages([]);
        setTimeout(() => {
          triggerReload(!reload);
        }, 1000);
      },
      onError: setError,
    });
  };

  const sendMessageToCurrentQueue = async (message: SqsMessage) => {
    let queueUrl = queues[tabIndex]?.QueueUrl || null;
    if (queueUrl !== null) {
      if (
        queues[tabIndex]?.QueueName.endsWith(".fifo") &&
        !message.messageGroupId
      ) {
        setError(
          "You need to set a MessageGroupID when sending Messages to a FIFO queue"
        );
        return;
      }
      await callApi({
        method: "POST",
        action: "SendMessage",
        queue: queues[tabIndex],
        message: message,
        onSuccess: () => {},
        onError: setError,
      });
    } else {
      setError("Could not send message to non-existent queue");
    }
  };

  return (
    <>
      <Container maxWidth="md" sx={{ marginY: "15px" }}>
        <Typography variant="h4">SQS Admin UI</Typography>
      </Container>
      <Container
        maxWidth="md"
        sx={{
          marginY: "15px",
          display: "grid",
          gap: 1,
          gridTemplateColumns: "repeat(4, 1fr)",
        }}
      >
        <CreateQueueDialog onSubmit={createNewQueue} />
        <Button
          variant="contained"
          disabled={disabledStatus}
          onClick={deleteCurrentQueue}
        >
          Delete Queue
        </Button>
        <SendMessageDialog
          disabled={disabledStatus}
          onSubmit={sendMessageToCurrentQueue}
          queue={queues[tabIndex]}
        />
        <Button
          variant="contained"
          disabled={disabledStatus}
          onClick={purgeCurrentQueue}
        >
          Purge Queue
        </Button>
      </Container>
      {error !== "" ? (
        <Container maxWidth="md">
          <Alert
            message={error}
            severity={"error"}
            onClose={() => setError("")}
          />
        </Container>
      ) : null}
      {queues?.length === 0 ? (
        <Container maxWidth="md">
          <MuiAlert severity="info">
            <AlertTitle>No Queue</AlertTitle>
            No Queues exist in region (default was "eu-central-1")
          </MuiAlert>
        </Container>
      ) : (
        <Container maxWidth="md">
          <Box sx={{ borderBottom: 1, borderColor: "divider" }}>
            <Tabs
              value={tabIndex}
              onChange={handleChange}
              aria-label="SQS overview"
            >
              {queues?.map((queue, index) => (
                <Tab label={queue.QueueName} {...a11yProps("tab", index)} />
              ))}
            </Tabs>
          </Box>
          {queues?.map((queue, index) => (
            <TabPanel
              value={tabIndex}
              index={index}
              {...a11yProps("tabpanel", index)}
            >
              <Grid container spacing={2}>
                {messages?.map((message, index) => (
                  <Grid item xs={12} {...a11yProps("gridItem", index)}>
                    <Paper>
                      <MessageItem
                        data={message}
                        {...a11yProps("messageItem", index)}
                      />
                    </Paper>
                  </Grid>
                ))}
              </Grid>
            </TabPanel>
          ))}
        </Container>
      )}
    </>
  );
};

export default Overview;
