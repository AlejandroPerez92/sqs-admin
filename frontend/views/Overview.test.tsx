import { render } from "@testing-library/react";
import Overview from "./Overview";
import { act } from "react";
import { AlertProvider } from "../components/AlertProvider";

describe("<Overview /> spec", () => {
  it("renders the Overview", async () => {
    let view;
    await act(async () => {
      view = render(
        <AlertProvider>
          <Overview />
        </AlertProvider>,
      );
    });
    expect(view).toMatchSnapshot();
  });
});
