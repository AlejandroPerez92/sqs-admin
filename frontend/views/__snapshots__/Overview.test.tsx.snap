// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overview /> spec renders the Overview 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-k008qs"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiDrawer-root MuiDrawer-docked css-1ct5g5z-MuiDrawer-docked"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1lwhjos-MuiPaper-root-MuiDrawer-paper"
              style="--Paper-shadow: none;"
            >
              <ul
                class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 css-1febv9v-MuiTypography-root"
                  >
                    SQS Admin UI
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-wyr59s-MuiTypography-root"
                  >
                    test
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-wyr59s-MuiTypography-root"
                  />
                </li>
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
                >
                  <div
                    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-11ylplz-MuiToolbar-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Create Queue
                    </button>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      Delete Queue
                    </button>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      Send message
                    </button>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      Purge Queue
                    </button>
                  </div>
                </li>
              </ul>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
              />
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
              />
              <ul
                class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1febv9v-MuiTypography-root"
              >
                Messages
              </h6>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthMd css-u4uvlt-MuiContainer-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-k1p3sh-MuiPaper-root-MuiAlert-root"
                role="alert"
                style="--Paper-shadow: none;"
              >
                <div
                  class="MuiAlert-icon css-vab54s-MuiAlert-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                    data-testid="InfoOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiAlert-message css-zioonp-MuiAlert-message"
                >
                  <div
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-5ick94-MuiTypography-root-MuiAlertTitle-root"
                  >
                    No Queue
                  </div>
                  No Queues exist in region: eu-central-1
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
          />
        </div>
      </div>
    </div>
    <div
      style="position: fixed; top: 20px; left: 50%; transform: translateX(-50%); z-index: 9999; width: 90%; max-width: 500px; pointer-events: none;"
    />
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-k008qs"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiDrawer-root MuiDrawer-docked css-1ct5g5z-MuiDrawer-docked"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1lwhjos-MuiPaper-root-MuiDrawer-paper"
            style="--Paper-shadow: none;"
          >
            <ul
              class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1febv9v-MuiTypography-root"
                >
                  SQS Admin UI
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-wyr59s-MuiTypography-root"
                >
                  test
                </h6>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-wyr59s-MuiTypography-root"
                />
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-11ylplz-MuiToolbar-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Create Queue
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    Delete Queue
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    Send message
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    Purge Queue
                  </button>
                </div>
              </li>
            </ul>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <ul
              class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1febv9v-MuiTypography-root"
            >
              Messages
            </h6>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthMd css-u4uvlt-MuiContainer-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-k1p3sh-MuiPaper-root-MuiAlert-root"
              role="alert"
              style="--Paper-shadow: none;"
            >
              <div
                class="MuiAlert-icon css-vab54s-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                  data-testid="InfoOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-zioonp-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-5ick94-MuiTypography-root-MuiAlertTitle-root"
                >
                  No Queue
                </div>
                No Queues exist in region: eu-central-1
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
